design-proposals/vm-console-logs.md

```diff
@@ -0,0 +1,743 @@
+ip-route6:unreachable default dev lo  metric -1  error -101
+ip-route6:ff00::/8 dev eth0  metric 256 
+ip-route6:unreachable default dev lo  metric -1  error -101
+=== datasource: nocloud local ===
+instance-id: 5a9fc181-957e-5c32-9e5a-2de5e9673531
+name: N/A
+availability-zone: N/A
+local-hostname: testvm
+launch-index: N/A
+=== cirros: current=0.4.0 uptime=3.71 ===
+  ____               ____  ____
+ / __/ __ ____ ____ / __ \/ __/
+/ /__ / // __// __// /_/ /\ \ 
+\___//_//_/  /_/   \____/___/ 
+   http://cirros-cloud.net
+
+
+login as 'cirros' user. default password: 'gocubsgo'. use 'sudo' for root.
+testvm login: stirabos@t14s:~$ 
+```
+
+Please notice that the serial console will remain available for the interactive use:
+```bash
+stirabos@t14s:~$ virtctl console -n simone testvm
+Successfully connected to testvm console. The escape sequence is ^]
+
+login as 'cirros' user. default password: 'gocubsgo'. use 'sudo' for root.
+testvm login: cirros
+Password: 
+$ 
+$ sudo sh -c "echo 'Hello, World!' >> /dev/console"
+Hello, World!
+$ 
+$ exit
+
+login as 'cirros' user. default password: 'gocubsgo'. use 'sudo' for root.
+testvm login: stirabos@t14s:~$ 
+stirabos@t14s:~$ 
+```
+
+and whatever will be written there will also be logged:
+```bash
+stirabos@t14s:~$ kubectl logs -n simone virt-launcher-testvm-t7h26 -c guest-console-log
+[    0.000000] Initializing cgroup subsys cpuset
+[    0.000000] Initializing cgroup subsys cpu
+[    0.000000] Initializing cgroup subsys cpuacct
+[    0.000000] Linux version 4.4.0-28-generic (buildd@lcy01-13) (gcc version 5.3.1 20160413 (Ubuntu 5.3.1-14ubuntu2.1) ) #47-Ubuntu SMP Fri Jun 24 10:09:13 UTC 2016 (Ubuntu 4.4.0-28.47-generic 4.4.13)
+...
+login as 'cirros' user. default password: 'gocubsgo'. use 'sudo' for root.
+testvm login: cirros
+Password: 
+$ 
+$ sudo sh -c "echo 'Hello, World!' >> /dev/console"
+Hello, World!
+$ 
+$ exit
+
+login as 'cirros' user. default password: 'gocubsgo'. use 'sudo' for root.
+testvm login: stirabos@t14s:~$ 
+```
+
+## API Examples
+
+On the VM CRD on `/spec/template/spec/domain/devices/autoattachSerialConsole` we already have:
+```yaml
+autoattachSerialConsole:
+  description: Whether to attach the default serial console
+    or not. Serial console access will not be available if
+    set to false. Defaults to true.
+  type: boolean
+```
+to let the VM owner easily attach the default serial console.
+We can introduce an additional boolean `/spec/template/spec/domain/devices/logSerialConsole`:
+```yaml
+logSerialConsole:
+  description: Whether to log the auto-attached default serial
+    console or not. Serial console logs will be collect to
+    a file and then streamed from a named 'guest-console-log'. Not
+    relevant if autoattachSerialConsole is disabled. Defaults
+    to false.
+  type: boolean
+```
+to let the VM owner require the serial console to be logged.
+It should be off by default since serial console logs could potentially disclosure sensitive data.
```

Comment:
+1
Might also negatively affect performance

Reply:
on https://github.com/kubevirt/kubevirt/pull/10110 I'm now trying to have it on by default for all the VMs and see if we can have it successfully passing the whole test suite with no evident regressions.